src/lib/storage/lz4_segment/lz4_encoder.hpp

```diff
@@ -168,35 +165,38 @@ class LZ4Encoder : public SegmentEncoder<LZ4Encoder> {
      * The offsets are stored as 32 bit unsigned integer as opposed to 64 bit (size_t) so that they can later be
      * compressed via vector compression.
      */
-    auto offsets = pmr_vector<uint32_t>{alloc};
-    offsets.resize(num_elements);
+    auto offsets = pmr_vector<uint32_t>{allocator};
```

Comment:
Shouldn't change anything. Binary caching still only works for values and dictionary segments, since the current benchmark do only support an initial encoding configuration, re-encoding doesn't happen anyways.

Reply:
What if the stored binary table is different from the requested one?

Reply:
Ahh, I see your point. That should work just fine now. I will check if there are some quirks.

Reply:
Works like a charm: https://gist.github.com/Bouncner/2d134821662626047cb8d629660e504c